## **Project: Analyzing The World Bank's international debt statistics**

<p>In this notebook, we are going to analyze international debt data collected by The World Bank. The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories. We are going to find the answers to questions like: </p>
<ul>
<li>What is the total amount of debt that is owed by the countries listed in the dataset?</li>
<li>Which country owns the maximum amount of debt and what does that amount look like?</li>
<li>What is the average amount of debt owed by countries across different debt indicators?</li>
</ul>
<p><img src="https://media-cldnry.s-nbcnews.com/image/upload/t_fit-1240w,f_auto,q_auto:best/newscms/2019_06/2746941/190208-stock-money-fanned-out-ew-317p.jpg" alt></p>

## **1.View of The World Bank's international debt data**
<p>The first line of code connects us to the <code>international_debt</code> database where the table <code>international_debt</code> is residing.

First, let's take a look at our data. Let's <code>SELECT</code> <em>all</em> of the columns from the <code>international_debt</code> table and limit the output to the first ten rows.</p>

In [ ]:
%%sql
postgresql:///international_debt

select *
from international_debt
limit 10

10 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('39107845')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('23779724.300000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('13335820')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('100847181.900000006')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('72894453.700000003'))]

## **2. Finding the number of distinct countries**
<p>From the first ten rows, we see that there are repetitions in the country names because a country is most likely to have debt in more than one debt indicator. </p>
<p>Without a count of unique countries, we will not be able to perform our statistical analyses holistically. In this section, we are going to extract the number of unique countries present in the table. </p>

In [ ]:
%%sql
SELECT count(distinct(country_name)) as total_distinct_countries
from international_debt

 * postgresql:///international_debt
1 rows affected.


[(124,)]

## **3. Finding out the distinct debt indicators**
<p>We can see there are a total of 124 countries present on the table.</p>

Knowing about the various debt indicators will help us to understand the areas in which a country can possibly be indebted to. </p>

to achieve that I'm using the <code>indicator_code</code> column which symbolizes the category of these debts.

In [ ]:
%%sql

SELECT
    DISTINCT indicator_code as distinct_debt_indicators
FROM
    international_debt
Order by distinct_debt_indicators

 * postgresql:///international_debt
25 rows affected.


[('DT.AMT.BLAT.CD',),
 ('DT.AMT.DLXF.CD',),
 ('DT.AMT.DPNG.CD',),
 ('DT.AMT.MLAT.CD',),
 ('DT.AMT.OFFT.CD',),
 ('DT.AMT.PBND.CD',),
 ('DT.AMT.PCBK.CD',),
 ('DT.AMT.PROP.CD',),
 ('DT.AMT.PRVT.CD',),
 ('DT.DIS.BLAT.CD',),
 ('DT.DIS.DLXF.CD',),
 ('DT.DIS.MLAT.CD',),
 ('DT.DIS.OFFT.CD',),
 ('DT.DIS.PCBK.CD',),
 ('DT.DIS.PROP.CD',),
 ('DT.DIS.PRVT.CD',),
 ('DT.INT.BLAT.CD',),
 ('DT.INT.DLXF.CD',),
 ('DT.INT.DPNG.CD',),
 ('DT.INT.MLAT.CD',),
 ('DT.INT.OFFT.CD',),
 ('DT.INT.PBND.CD',),
 ('DT.INT.PCBK.CD',),
 ('DT.INT.PROP.CD',),
 ('DT.INT.PRVT.CD',)]

## 4. **Totaling the amount of debt owed by the countries**

<p>Let's switch gears from the debt indicators now and find out the total amount of debt (in USD) that is owed by all the different countries. This will give us a sense of how the overall economy of the entire world is holding up.</p>

**Note**: we divide the total debt by 1000000 and round the result to 2 decimal places so that the output is fathomable.

In [ ]:
%%sql

SELECT
    round(sum(debt)/1000000,2) as total_debt
FROM
    international_debt

 * postgresql:///international_debt
1 rows affected.


[(Decimal('3079734.49'),)]

## **5. Country with the highest debt**
<p>"Human beings cannot comprehend very large or very small numbers. It would be useful for us to acknowledge that fact." - <a href="https://en.wikipedia.org/wiki/Daniel_Kahneman">Daniel Kahneman</a>. That is more than <em>3 million <strong>million</strong></em> USD, an amount which is really hard for us to fathom. </p>
<p>Now that we have the exact total of the amounts of debt owed by several countries, let's now find out the country that owns the highest amount of debt along with that amount.

<strong>Note</strong> : This debt is the sum of different debts owed by a country across several categories.

In [ ]:
%%sql

SELECT country_name, ROUND5debt)/ 1000000as total_debt
FROM international_debt
group by country_name
Order by total_debt desc
limit 1

 * postgresql:///international_debt
1 rows affected.


[('China', Decimal('285793494734.200001568'))]

## **6. Average amount of debt across indicators**
<p>So, it was <em>China</em>. We now have a brief overview of the dataset and a few of its summary statistics.

We already have an idea of the different debt indicators in which the countries owe their debts. Now, let's dig even further to get a better sense of the distribution of the amount of debt across different indicators through exploring **how much debt on an average countries owe across these indicators**.

In [ ]:
%%sql
SELECT
    indicator_code as debt_indicator,
    indicator_name,
    round(avg(debt)/1000000,2) as average_debt
FROM international_debt
group by debt_indicator, indicator_name
ORDER BY average_debt desc
limit 10

 * postgresql:///international_debt
10 rows affected.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('5904.87')),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('5161.19')),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', Decimal('2152.04')),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('1958.98')),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('1803.69')),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', Decimal('1644.02')),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', Decimal('1223.14')),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('1220.41')),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('1191.19')),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', Decimal('1082.62'))]

## **7. The highest amount of principal repayments**
<p>We can see that the indicator <code>DT.AMT.DLXF.CD</code> tops the chart of average debt. This category includes repayment of long term debts. Countries take on long-term debt to acquire immediate capital. More information about this category can be found <a href="https://datacatalog.worldbank.org/principal-repayments-external-debt-long-term-amt-current-us-0">here</a>. </p>
<p>An interesting observation in the above finding is that there is a huge difference in the amounts of the indicators after the second one. This indicates that the first two indicators might be the most severe categories in which the countries owe their debts.</p>
<p>We can investigate this a bit more so as to find out which country owes the highest amount of debt in the category of long term debts (<code>DT.AMT.DLXF.CD</code>). Since not all the countries suffer from the same kind of economic disturbances, this finding will allow us to understand that particular country's economic condition a bit more specifically. </p>

In [ ]:
%%sql
SELECT
    country_name,
    indicator_name
from international_debt
where debt = (select max(debt)
                         from international_debt
                         where indicator_code = 'DT.AMT.DLXF.CD')

 * postgresql:///international_debt
1 rows affected.


[('China', 'Principal repayments on external debt, long-term (AMT, current US$)')]

## **8. The most common debt indicator**
<p>China has the highest amount of debt in the long-term debt (<code>DT.AMT.DLXF.CD</code>) category. This is verified by <a href="https://data.worldbank.org/indicator/DT.AMT.DLXF.CD?end=2018&most_recent_value_desc=true">The World Bank</a>. It is often a good idea to verify our analyses like this since it validates that our investigations are correct. </p>
<p>We saw that long-term debt is the topmost category when it comes to the average amount of debt. But is it the most common indicator in which the countries owe their debt? Let's find that out. </p>

In [ ]:
%%sql
SELECT
    indicator_code,
    count(indicator_code) as indicator_count
FROM international_debt
group by indicator_code
Order by indicator_count desc, indicator_code desc
limit 20

 * postgresql:///international_debt
20 rows affected.


[('DT.INT.OFFT.CD', 124),
 ('DT.INT.MLAT.CD', 124),
 ('DT.INT.DLXF.CD', 124),
 ('DT.AMT.OFFT.CD', 124),
 ('DT.AMT.MLAT.CD', 124),
 ('DT.AMT.DLXF.CD', 124),
 ('DT.DIS.DLXF.CD', 123),
 ('DT.INT.BLAT.CD', 122),
 ('DT.DIS.OFFT.CD', 122),
 ('DT.AMT.BLAT.CD', 122),
 ('DT.DIS.MLAT.CD', 120),
 ('DT.DIS.BLAT.CD', 113),
 ('DT.INT.PRVT.CD', 98),
 ('DT.AMT.PRVT.CD', 98),
 ('DT.INT.PCBK.CD', 84),
 ('DT.AMT.PCBK.CD', 84),
 ('DT.INT.DPNG.CD', 79),
 ('DT.AMT.DPNG.CD', 79),
 ('DT.INT.PBND.CD', 69),
 ('DT.AMT.PBND.CD', 69)]

## **9. Other viable debt issues and conclusion**
<p>There are a total of six debt indicators in which all the countries listed in our dataset have taken debt. The indicator <code>DT.AMT.DLXF.CD</code> is also there in the list. So, this gives us a clue that all these countries are suffering from a common economic issue. But that is not the end of the story, but just a part of the story.</p>
<p>Let's change tracks from <code>debt_indicator</code>s now and focus on the amount of debt again. Let's find out the maximum amount of debt that each country has. With this, we will be in a position to identify the other plausible economic issues a country might be going through.</p>


In [ ]:
%%sql
SELECT
    country_name,
    round(max(debt)/1000000,2) as maximum_debt
from international_debt
GROUP BY country_name
order by maximum_debt desc
limit 10

 * postgresql:///international_debt
10 rows affected.


[('China', Decimal('96218.62')),
 ('Brazil', Decimal('90041.84')),
 ('Russian Federation', Decimal('66589.76')),
 ('Turkey', Decimal('51555.03')),
 ('South Asia', Decimal('48756.30')),
 ('Least developed countries: UN classification', Decimal('40160.77')),
 ('IDA only', Decimal('34531.19')),
 ('India', Decimal('31923.51')),
 ('Indonesia', Decimal('30916.11')),
 ('Kazakhstan', Decimal('27482.09'))]

## **Summary**
<p>In this notebook, we took a look at debt owed by countries across the globe. We extracted a few summary statistics from the data and unraveled some interesting facts and figures. We also validated our findings to make sure the investigations are correct.</p>